### PR TITLE
Mark deleted branche builds as failed (fixes #243)

### LINKED ISSUE #243: Handle branch deleted event
If there are unstarted builds with a branch when it is deleted the build should be marked as errored.

### PR DESCRIPTION
**Pull request by frecar**
@relekang

### CODE REVIEW

File: frigg/webhooks/events/github.py [hunk 1]
```diff
@@ -34,12 +34,14 @@ def repository_private(self):
 
     @property
     def branch(self):
-        if self.event_type == 'push':
+        if self.event_type in 'push':
```
Comment (relekang) [inline frigg/webhooks/events/github.py]: Why change this?

Comment (frecar) [inline frigg/webhooks/events/github.py]: typo! 2sec

File: frigg/webhooks/test_views.py [hunk 2]
```diff
@@ -131,6 +131,21 @@ def test_push_delete(self, mock_start_build):
...
+        Build.objects.get_or_create(
+            project=project,
+            branch="frecar/add-worker-host"
+        )
+
+        response = self.post_webhook('delete', fixture='push_delete_branch.json')
+        self.assertStatusCode(response)
+        self.assertFalse(mock_start_build.called)
+
+        project = Project.objects.get_or_create(owner='frigg', name='frigg-hq', private=False)[0]
```
Comment (relekang) [inline frigg/webhooks/test_views.py]: Is it better to only use a `get()` here? I guess the test should fail if the project has been deleted :)

Comment (frecar) [inline frigg/webhooks/test_views.py]: agreed

**Comment by relekang** [comment]
LGTM :sparkles:

**Pull request merged**